### PR TITLE
[ioctl] modify client  interface  APIServiceClient(APIServiceConfig) to  APIServiceClient()

### PR DESCRIPTION
**Pull request by huof6829**
# Description

argument of endpoint and insecure is set when create command,  so delete argument  for simple using.   

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/client.go [hunk 1]
```diff
@@ -92,6 +92,11 @@ type (
 	}
 )
 
+var (
+	// APIServiceCfg represents the persistent flags of cobra command and will be parsed after rootCmd.Execute() in main
+	APIServiceCfg = APIServiceConfig{Insecure: false}
```
Comment (Liuhaai) [inline ioctl/client.go:97]: Why is global var used here?

Comment (Liuhaai) [inline ioctl/client.go:97]: IMO, global var shouldn't be used here

Comment (huof6829) [reply ioctl/client.go]: modify to function

File: ioctl/newcmd/account/account.go [hunk 2]
```diff
@@ -89,8 +85,8 @@ func NewAccountCmd(client ioctl.Client) *cobra.Command {
 	_flagEndpointUsage, _ := client.SelectTranslation(_flagEndpoint)
 	_flagInsecureUsage, _ := client.SelectTranslation(_flagInsecure)
 
-	ac.PersistentFlags().StringVar(&endpoint, "endpoint", client.Config().Endpoint, _flagEndpointUsage)
-	ac.PersistentFlags().BoolVar(&insecure, "insecure", !client.Config().SecureConnect, _flagInsecureUsage)
+	ac.PersistentFlags().StringVar(client.Endpoint(), "endpoint", client.Config().Endpoint, _flagEndpointUsage)
```
Comment (Liuhaai) [inline ioctl/newcmd/account/account.go:88]: add `todo` to move it to maincmd

**Review by Liuhaai — APPROVED** [review]

File: ioctl/newcmd/account/account.go (hunk 2 above)
Comment (huof6829) [reply ioctl/newcmd/account/account.go:88]: alias/contract/update  these sub commands donot support `--endpoint  --inseure`.  Separately set will be better, i think

Comment (Liuhaai) [reply ioctl/newcmd/account/account.go:88]: OK good point

File: ioctl/client.go [hunk 3]
```diff
@@ -39,8 +39,12 @@ type (
 		Stop(context.Context) error
 		// Config returns the config of the client
 		Config() config.Config
+		// Endpoint returns the address of the member to receive input flag value
+		Endpoint() *string
```
Comment (CoderZhi) [inline ioctl/client.go:43]: if you want to receive input flag value, returning a pointer is not a good way. 
`SetEndpointWithFlag(cb func(...))`, in which `cb` is a callback function, whose value is `StringVar` may be better

File: ioctl/client.go [hunk 4]
```diff
@@ -39,8 +39,12 @@ type (
...
+		// Insecure returns the address of the member to receive input flag value
+		Insecure() *bool
```
Comment (CoderZhi) [inline ioctl/client.go:45]: same here

File: ioctl/client.go (hunk 3 above)
Comment (huof6829) [reply ioctl/client.go:43]: I will try to modify. Thanks.

Comment (huof6829) [reply ioctl/client.go]: It's OK, Good idea !  I'v done.

File: ioctl/client.go (hunk 4 above)
Comment (huof6829) [reply ioctl/client.go]: It's OK, Good idea!  I'v done.

File: ioctl/newcmd/account/account.go [hunk 5]
```diff
@@ -88,10 +84,8 @@ func NewAccountCmd(client ioctl.Client) *cobra.Command {
 
 	_flagEndpointUsage, _ := client.SelectTranslation(_flagEndpoint)
 	_flagInsecureUsage, _ := client.SelectTranslation(_flagInsecure)
-
-	ac.PersistentFlags().StringVar(&endpoint, "endpoint", client.Config().Endpoint, _flagEndpointUsage)
-	ac.PersistentFlags().BoolVar(&insecure, "insecure", !client.Config().SecureConnect, _flagInsecureUsage)
-
+	client.SetEndpointWithFlag(ac.PersistentFlags().StringVar, _flagEndpointUsage)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/account.go:87]: if you want to create this flag inside of client, passing `ac.PersistentFlags()` to client via `CreateFlagVars`.
Also, _flagEndpointUsage and _flagInsecureUsage are not command dependent, which should be defined in client

Comment (huof6829) [reply ioctl/newcmd/account/account.go]: I see, thx. 
I'v done!

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**